src/Form/MessageTemplateForm.php

```diff
@@ -170,16 +171,18 @@ public static function addMoreAjax(array $form, FormStateInterface $form_state)
    * {@inheritdoc}
    */
   public function save(array $form, FormStateInterface $form_state) {
-    $values = $form_state->getValue('text');
-    usort($values, 'message_order_text_weight');
-
-    // Saving the message text values.
-    foreach ($values as $key => $value) {
-      $values[$key] = $value['value'];
-    }
-
-    $this->entity->set('text', $values);
-    $this->entity->save();
+    // Sort by weight.
+    $text = $form_state->getValue('text');
+    usort($text, function ($a, $b) {
+      return SortArray::sortByKeyInt($a, $b, '_weight');
+    });
+    // Do not store weight, as these are now sorted.
+    $text = array_map(function ($a) {
+      unset($a['_weight']); return $a;
```

Comment:
is this a common practice? What's the down side of keeping the weight

(just asking, sorting before saving looks fine)